src/main/java/io/confluent/connect/jdbc/JdbcSinkConnector.java

```diff
@@ -65,8 +67,25 @@ public ConfigDef config() {
 
   @Override
   public Config validate(Map<String, String> connectorConfigs) {
-    // TODO cross-fields validation here: pkFields against the pkMode
-    return super.validate(connectorConfigs);
+    /** get configuration parsed and validated individually */
+    Config config = super.validate(connectorConfigs);
+
+    config.configValues()
+        .stream()
+        .filter(cfg -> PK_MODE.equals(cfg.name()))
+        .findFirst()
+        /**
+         * use recommended values for configuration validation
+         * {@link PrimaryKeyModeRecommender#validValues}
+         */
+        .ifPresent(pkMode -> {
+          if (!pkMode.recommendedValues().contains(pkMode.value())) {
+            pkMode.addErrorMessage("'" + pkMode.value() + "' is invalid:"
+                + " Deletes are only supported for pk.mode 'record_key'");
+          }
```

Comment:
Is this susceptible to false positives if the user gives a value for the `pk.mode` property that is invalid not because of conflict with `delete.enabled` but just because it doesn't correspond to a recognized primary key mode? For example, something like `pk.mode = not_a_real_mode`.

Reply:
that is great catch! thank you.